### PR TITLE
Mail service

### PR DESCRIPTION
**Pull request by mkkasem**
created send mail function and added test file

### CODE REVIEW

**Comment by Shrreya** [comment]
@mehmetfatiherdem Would you be able to do the first review on this PR? After you, I can add in my review.

File: src/services/__tests__/mail.test.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    });
+    expect(spyOnSendMail).toHaveBeenCalled();
+    expect(spyOnSendMail).toHaveBeenCalledTimes(1);
+    expect(spyOnSendMail).toHaveBeenCalledWith(emailOptions);
+    expect(spyOnSendMail).not.toThrow();
+    expect(res).toEqual(sendMailResult);
+    spyOnSendMail.mockRestore();
+  });
+
+  test('Should throw an error if the there are no recipients', async () => {
```
Comment (mehmetfatiherdem) [inline src/services/__tests__/mail.test.js:48]: You should remove the `the` before `there`. You can just keep it as `Should throw an error if there are no recipients`

File: src/services/__tests__/mail.test.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    expect(spyOnCreateTransport).toHaveBeenCalled();
+    expect(spyOnCreateTransport).toHaveBeenCalledTimes(1);
+    expect(spyOnCreateTransport).toHaveReturned();
+    expect(spyOnSendMail).toHaveBeenCalled();
+    expect(spyOnSendMail).toHaveBeenCalledWith(emailOptions);
+    expect(spyOnSendMail).rejects.toThrow('No recipients');
+    spyOnSendMail.mockRestore();
+  });
+
+  // NOTE: I think this test is not necessary because invalid emails should be detected when user signs up
```
Comment (mehmetfatiherdem) [inline src/services/__tests__/mail.test.js:72]: I aggree with you on this case. However, there may be a case where an email address is actually pass the validity checks and still it is not valid. It may be deleted, suspended or somehow disabled before signing up or maybe after a while they signed up. So, maybe we can add a general test case for sending email failed.

File: src/services/mail.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  sendEmail: async (emailOptions) => {
+    try {
+      const transporter = nodemailer.createTransport({
+        host: 'smtp.office365.com',
+        auth: {
+          user: process.env.EMAIL_USER,
+          pass: process.env.EMAIL_PASSWORD,
+        },
+      });
+      const res = await transporter.sendMail(emailOptions);
```
Comment (mehmetfatiherdem) [inline src/services/mail.js:20]: Regarding the suggestion above, I think you can add the second argument callback to fetch errors here in the `sendMail` function.

**Review by mehmetfatiherdem — CHANGES_REQUESTED** [review]
Good job with the mail functionality :clap:. I added a few suggestion to you.

File: src/services/__tests__/mail.test.js (hunk 2 above)
Comment (Shrreya) [reply src/services/__tests__/mail.test.js:72]: I agree with @mehmetfatiherdem 
This is a unit test for the `sendMail` function, this unit test does not know if another functionality with email validation exists. So it should check for all possible edge cases without assuming it's being tested somewhere else.

We can keep this test case as it, and you can remove the above comment.

File: src/services/mail.js (hunk 3 above)
Comment (Shrreya) [reply src/services/mail.js:20]: @mehmetfatiherdem Good suggestion! Callbacks is one approach to catch errors, here @mkkasem has used a try/catch block so errors will be checked in the catch block.

File: src/services/mail.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      return res;
+    } catch (error) {
+      return error;
```
Comment (Shrreya) [inline src/services/mail.js:23]: It will be good to add an error log here so that if any email send errors take place when frontend team integrates with the API, we will have a record of it in our log files.

**Review by Shrreya — CHANGES_REQUESTED** [review]
@mkkasem Brilliant work! And I'm so happy to see such well written tests.

Just a few small suggestions below and then we can merge this.

File: src/services/mail.js (hunk 3 above)
Comment (mkkasem) [reply src/services/mail.js:21]: yes, we discussed it during the standup.

**Review by Shrreya — APPROVED** [review]

**Comment by mkkasem** [comment]
@mehmetfatiherdem  @AmmarBaki2  @Peri7at  guys can anyone approve this PR to be able to merge it?

**Review by mehmetfatiherdem — APPROVED** [review]
